anndata/_core/merge.py

```diff
@@ -14,6 +14,7 @@
     Collection,
     Iterable,
     Optional,
+    Sequence,
```

Comment:
`typing.Sequence` is deprecated in favor of `collections.abc.Sequence`: https://docs.python.org/3/library/typing.html#typing.Sequence

Reply:
I thought you didn’t like unrelated changes, but sure.

Reply:
I would also be okay with not doing the typing changes as well. But this import was added in this PR, no?